src/execution/compiler/operator/index_create_translator.cpp

```diff
@@ -282,7 +282,10 @@ void IndexCreateTranslator::IndexInsert(WorkContext *ctx, FunctionBuilder *funct
                                           codegen_->ConstBool(index_schema.Unique())});
   auto *cond = codegen_->UnaryOp(parsing::Token::Type::BANG, index_insert_call);
   If success(function, cond);
-  { function->Append(codegen_->AbortTxn(GetExecutionContext())); }
+  {
+    TearDownStorageInterface(function, local_storage_interface_.GetPtr(codegen_));
```

Comment:
Seems like you may have forgotten to update this class to move this logic to the pipeline state.

Reply:
weirdly enough, this already had it in the pipeline state. I removed my change, and it looks like things are fine:

```
struct QueryState {
    execCtx        : *ExecutionContext
    global_col_oids: [1]uint32
}
struct P1_State {
    local_storage_interface: StorageInterface
    local_index_pr         : *ProjectedRow
    local_tuple_slot       : TupleSlot
    num_inserts            : uint32
    execFeatures           : ExecOUFeatureVector
}
fun Query0_Init(queryState: *QueryState) -> nil {
    queryState.global_col_oids[0] = 1
    return
}

fun Query0_Pipeline1_PostHook(queryState: *QueryState, pipelineState: *P1_State, dummyArg: *nil) -> nil {
    @execOUFeatureVectorInit(queryState.execCtx, &pipelineState.execFeatures, 1, true)
    @registerThreadWithMetricsManager(queryState.execCtx)
    @execCtxStartPipelineTracker(queryState.execCtx, 1)
    var num_tuples = @indexGetSize(&pipelineState.local_storage_interface)
    @execOUFeatureVectorRecordFeature(&pipelineState.execFeatures, 1, 10000, 0, 0, num_tuples)
    @execOUFeatureVectorRecordFeature(&pipelineState.execFeatures, 1, 10000, 1, 0, num_tuples)
    var heap_size = @storageInterfaceGetIndexHeapSize(&pipelineState.local_storage_interface)
    @execCtxSetMemoryUseOverride(queryState.execCtx, heap_size)
    @execCtxEndPipelineTracker(queryState.execCtx, 0, 1, &pipelineState.execFeatures)
    @aggregateMetricsThread(queryState.execCtx)
    @execOUFeatureVectorReset(&pipelineState.execFeatures)
    return
}

fun Query0_Pipeline1_InitPipelineState(queryState: *QueryState, pipelineState: *P1_State) -> nil {
    @storageInterfaceInit(&pipelineState.local_storage_interface, queryState.execCtx, 1061, queryState.global_col_oids, false)
    pipelineState.local_index_pr = @getIndexPR(&pipelineState.local_storage_interface, 1062)
    return
}

fun Query0_Pipeline1_TearDownPipelineState(queryState: *QueryState, pipelineState: *P1_State) -> nil {
    @storageInterfaceFree(&pipelineState.local_storage_interface)
    @execOUFeatureVectorReset(&pipelineState.execFeatures)
    return
}

fun Query0_Pipeline1_ParallelWork(queryState: *QueryState, pipelineState: *P1_State, tvi: *TableVectorIterator) -> nil {
    pipelineState.num_inserts = 0
    @execOUFeatureVectorInit(queryState.execCtx, &pipelineState.execFeatures, 1, false)
    @registerThreadWithMetricsManager(queryState.execCtx)
    @execCtxStartPipelineTracker(queryState.execCtx, 1)
    for (@tableIterAdvance(tvi)) {
        var vpi = @tableIterGetVPI(tvi)
        for (; @vpiHasNext(vpi); @vpiAdvance(vpi)) {
            pipelineState.local_tuple_slot = @vpiGetSlot(vpi)
            @prSetIntNull(pipelineState.local_index_pr, 0, @vpiGetIntNull(vpi, 0))
            if (!@indexInsertWithSlot(&pipelineState.local_storage_interface, &pipelineState.local_tuple_slot, false)) {
                @abortTxn(queryState.execCtx)
            }
            pipelineState.num_inserts = pipelineState.num_inserts + IntegralCast(1)
        }
    }
    @execCtxSetMemoryUseOverride(queryState.execCtx, 0)
    @execOUFeatureVectorRecordFeature(&pipelineState.execFeatures, 1, 10000, 0, 0, pipelineState.num_inserts)
    @execOUFeatureVectorRecordFeature(&pipelineState.execFeatures, 1, 10000, 1, 0, pipelineState.num_inserts)
    @execCtxEndPipelineTracker(queryState.execCtx, 0, 1, &pipelineState.execFeatures)
    @aggregateMetricsThread(queryState.execCtx)
    @execOUFeatureVectorReset(&pipelineState.execFeatures)
    return
}

fun Query0_Pipeline1_Init(queryState: *QueryState) -> nil {
    var threadStateContainer = @execCtxGetTLS(queryState.execCtx)
    @tlsReset(threadStateContainer, @sizeOf(P1_State), Query0_Pipeline1_InitPipelineState, Query0_Pipeline1_TearDownPipelineState, queryState)
    return
}

fun Query0_Pipeline1_Run(queryState: *QueryState) -> nil {
    var pipelineState = @ptrCast(*P1_State, @tlsGetCurrentThreadState(@execCtxGetTLS(queryState.execCtx)))
    @execCtxInitHooks(queryState.execCtx, 1)
    @execCtxRegisterHook(queryState.execCtx, 0, Query0_Pipeline1_PostHook)
    @iterateTableParallel(1061, queryState.global_col_oids, queryState, queryState.execCtx, Query0_Pipeline1_ParallelWork)
    @execCtxClearHooks(queryState.execCtx)
    return
}

fun Query0_Pipeline1_TearDown(queryState: *QueryState) -> nil {
    @tlsClear(@execCtxGetTLS(queryState.execCtx))
    @ensureTrackersStopped(queryState.execCtx)
    return
}

fun Query0_TearDown(queryState: *QueryState) -> nil {
    return
}
```